### PR TITLE
Fix "icon" parameter in EditorPlugin.add_custom_type is not optional but doc says it is

### PR DESCRIPTION
**Pull request by ashish0kumar**
fixed an error in doc/classes/EditorPlugin.xml: "icon" parameter in EditorPlugin.add_custom_type is not optional, but documentation says it is.
Removed the line "An icon can be optionally passed." and added this "An editor error is generated when this function is called without the icon parameter."

### CODE REVIEW

File: doc/classes/EditorPlugin.xml [hunk 1]
```diff
@@ -444,7 +444,7 @@
 			<param index="2" name="script" type="Script" />
 			<param index="3" name="icon" type="Texture2D" />
 			<description>
-				Adds a custom type, which will appear in the list of nodes or resources. An icon can be optionally passed.
+				Adds a custom type, which will appear in the list of nodes or resources. An editor error is generated when this function is called without the icon parameter.
```
Comment (AThousandShips) [inline doc/classes/EditorPlugin.xml:447]:
```suggestion
				Adds a custom type, which will appear in the list of nodes or resources.
```
Better to just remove, unless it should say that the icon can be `null`, but this description is unnecessary

**Comment by Mickeon** [comment]
Good catch, although it _really_ should be optional, I don't see why it isn't.

**Review by Mickeon — APPROVED** [review]

**Comment by AThousandShips** [comment]
Please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Review by AThousandShips — APPROVED** [review]

**Review by Mickeon — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**